napi_from_js.md

```diff
@@ -0,0 +1,367 @@
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_cb_data*
+
+#### Signature
+```C
+napi_status napi_get_cb_data(napi_env env,
+                             napi_callback_info cbinfo,
+                             void** result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  cbinfo`: The callback info passed into the callback function
+- `[out] result`: The pointer to the `data` parameter that was passed
+in when the callback was initially installed
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### Description
+This method allows you to retrieve the data pointer that you had passed in to 
+[`napi_create_function`](#napi_create_function) when you originally created
+the function object.
+
+### *napi_get_cb_this*
+
+#### Signature
+```C
+napi_status napi_get_cb_this(napi_env env,
+                             napi_callback_info cbinfo,
+                             napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  cbinfo`: The callback info passed into the callback function
+- `[out] result`: The `napi_value` representing the value of `this` 
+provided for the call to the function.
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This method retrieves the `this` pointer that was passed in when the add-on's
+native method was invoked from JavaScript.
+
+### *napi_get_cb_info*
+
+#### Signature
+```C
+napi_status napi_get_cb_info(napi_env env,
+                             napi_callback_info cbinfo,
+                             size_t* argc, 
+                             napi_value* argv,
+                             napi_value* thisArg, 
+                             void** data)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  cbinfo`: The callback info passed into the callback function
+- `[in-out] argc`: Specifies the size of the provided `argv` array
+and receives the actual count of arguments.
+- `[out] argv`: Buffer to which the `napi_value` representing the
+arguments are copied
+- `[out] this`: Receives the JS `this` argument for the call
+- `[out] data`: Receives the data pointer for the callback.
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This is a convenience method that retrieves all relevant information from a 
+`napi_callback_info` in a single call.
+
+### *napi_is_construct_call*
+
+#### Signature
+```C
+napi_status napi_is_construct_call(napi_env env,
+                                   napi_callback_info cbinfo,
+                                   bool* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  cbinfo`: The callback info passed into the callback function
+- `[out] result`: Whether the native function is being invoked as 
+a constructor call
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
```

Comment:
Yeah, that was my thought too - I left it as is for now, let me know if you feel strongly about this